src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,455 @@
+            return returnsAPromise(node, checker) || (isCallExpression(node) && (isCallback(node, "then", checker) || isCallback(node, "catch", checker)));
+        }
+
+        forEachChild(func.body, function visit(node: Node) {
+            if (isCallExpression(node) && isCallback(node, "then", checker)) {
+                lastDotThen.set(String(getNodeId(node)), false);
+
+                for (const arg of node.arguments) {
+                    forEachChild(arg, function visit(argChild: Expression) {
+                        if (willBeParsed(argChild)) {
+                            lastDotThen.set(String(getNodeId(argChild)), false);
+                        }
+                    });
+                }
+
+                forEachChild(node, function visit(child: Node) {
+                    if (isExpression(child) && willBeParsed(child)) {
+                        if (!lastDotThen.get(String(getNodeId(child)))) {
+                            lastDotThen.set(String(getNodeId(child)), true);
+                        }
+
+                        if (!isCallExpression(child)) {
+                            return;
+                        }
+
+                        for (const arg of child.arguments) {
+                            forEachChild(arg, function visit(argChild: Expression) {
+                                if (willBeParsed(argChild)) {
+                                    lastDotThen.set(String(getNodeId(argChild)), true);
+                                }
+                            });
+                        }
+                    }
+
+                    forEachChild(child, visit);
+                });
+            }
+            else {
+                forEachChild(node, visit);
+            }
+        });
+    }
+
+    // varNamesMap holds all of the variables in original source code. synthNamesMap holds all of the variables created by the refactor
+    function renameCollidingVarNames(nodeToRename: FunctionLikeDeclaration, checker: TypeChecker, synthNamesMap: Map<[Identifier, number]>, context: CodeFixContextBase): FunctionLikeDeclaration {
+        const allVarNames: Identifier[] = [];
+
+        function isFunctionRef(node: Node): boolean {
+            const callExpr = climbPastPropertyAccess(node);
+            return !isCallExpression(callExpr) || callExpr.expression !== node;
+        }
+
+        function definedInFile(symbol: Symbol, sourceFile: SourceFile): boolean {
+            return symbol.valueDeclaration && symbol.valueDeclaration.getSourceFile() === sourceFile;
+        }
+
+        forEachChild(nodeToRename, function visit(node: Node) {
+
+            if (isIdentifier(node)) {
+                const type = checker.getTypeAtLocation(node);
+                const symbol = checker.getSymbolAtLocation(node);
+                const newName = getNewNameIfConflict(node, allVarNames);
+
+                if (symbol && !node.symbol) {
+                    node.symbol = symbol;
+                }
+

```

Comment:
Nothing happens after this line unless `definedInFile(symbol, context.sourceFile)` is true, so we can check that exactly once and bail early instead of checking at both `if`s